### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by jsafrane**
* Bump `govc` to the latest released version.
* Add XMLs with 1 host, 2 node vSphere cluster to `pkg/check/testdata/default` - that's how a simulated vCenter will look like. Added README there to be able to re-generate it when needed, with special attention to simulated datastore (it needs a real directory on filesystem).
* Add a simple test framework.
  * To start the simulator (loading the XMLs from filesystem) and serving a https endpoint.
  * To modify simulated vCenter hosts+VMs to test various scenarios.
  * To mock KubeClient with dummy API objects.
* Add unit tests for most checks.
  * `folder_test.go` does not have any error cases, we would need to simulate permission errors.
  * No tests for `tasks.go`, did not find a way how to inject a task to the simulator.
  * No test for StorageClass with StoragePolicy - I'd need to add a few simulated objects. Deferring to subsequent PRs.
 
@openshift/storage

### CODE REVIEW

File: pkg/check/testdata/default/README [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Generated using vcsim and vcgo, both v0.23.1.
```
[the remark this answers is not in the document]
Comment (gnufied) [reply pkg/check/testdata/default/README:1]: This also reads bit weird - it looks like as if this readme was generated.

**Comment by gnufied** [comment]
lets merge this. we can fix readme in a follow up.

/lgtm

**Comment by jsafrane** [comment]
rebased

**Pull request merged**